src/Sulu/Component/Content/PropertyParameter.php

```diff
@@ -0,0 +1,213 @@
+<?php
+/*
+ * This file is part of the Sulu CMF.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content;
+
+use JMS\Serializer\Annotation\Exclude;
+use JMS\Serializer\Annotation\HandlerCallback;
+use JMS\Serializer\Annotation\Type;
+use JMS\Serializer\Context;
+use JMS\Serializer\JsonDeserializationVisitor;
+use JMS\Serializer\JsonSerializationVisitor;
+use JMS\Serializer\Metadata\PropertyMetadata;
+use JMS\Serializer\Metadata\StaticPropertyMetadata;
+
+/**
+ * Represents a parameter of a property
+ */
+class PropertyParameter
+{
+    /**
+     * @var string
+     * @Type("string")
+     */
+    private $name;
+
+    /**
+     * @var string|bool|array
+     * @Type("array<string,Sulu\Component\Content\PropertyParameter>")
+     */
+    private $value;
+
+    /**
+     * @var string
+     * @Type("string")
+     */
+    private $type;
+
+    /**
+     * @var Metadata
+     * @Type("Sulu\Component\Content\Metadata")
+     */
+    private $metadata;
+
+    /**
+     * Constructor
+     * @param string $name
+     * @param string $type
+     * @param string|bool|array $value
+     * @param array $metadata
+     */
+    function __construct($name, $type, $value, $metadata = array())
```

Comment:
Wouldn't it be more comfortable to have the order `$name, $value, $type, $metadata`, and default type to null? So that it is only set if needed? I also would not use the type string, but for special stuff like collections.